### PR TITLE
Notify maintainers when dependency is out of range

### PR DESCRIPTION
**Pull request by ysangkok**
Tested by:
1. lowering limits as explained in `UserNotify.hs` to 1 minute
2. installing postfix, configuring it for "Localhost delivery only" and changing my Hackage user's email to `janus@myhostname.local.domain`
3. enable the notification on the preferences page, opt in to notifications
4. uploading transformers-0.5.6.2 (with patches to fix stricter cabal file validation)
5. uploading servant-0.19.1 (which doesn't accept transformers-0.6)
6. uploading transformers-0.6.0.6
7. after a minute, verified using `mail(1)` from the package `bsd-mailx` that the mail was delivered and the link works

### CODE REVIEW

**Comment by mitchellwrosen** [comment]
This would be an awesome feature! :tada:

File: src/Distribution/Server/Features/UserNotify.hs [hunk 1]
```diff
@@ -357,12 +453,109 @@ initUserNotifyFeature env@ServerEnv{ serverStateDir, serverTemplatesDir,
...
+    Just foundPackage -> do
+      let
+        vertices :: Set.Set Vertex
+        vertices = suc revs foundPackage
+        revDepNames :: [PackageName]
+        revDepNames = mapMaybe (`lookupR` nodemap) (Set.toList vertices)
+      toNotify <- traverse maintainersToNotify revDepNames
+      pure $
+        Map.fromList
+          [ ( maintainerId
```
Comment (aaronallen8455) [inline src/Distribution/Server/Features/UserNotify.hs:487]: What happens here if multiple packages by the same maintainer are affected by the same release? Ideally, I think the maintainer should get a single notification with a list of those packages.

Comment (ysangkok) [reply src/Distribution/Server/Features/UserNotify.hs]: Ok, I have pushed a commit which groups by userId and packageId. Now the email looks like this:

```
The dependency transformers-0.6.0.2 has been updated.

You have requested to be notified when a dependency isn't accepted by any of your maintained packages.

These are your packages that require transformers but don't accept 0.6.0.2:

servant-0.19.1

You can adjust your notification preferences at

http://gorm.local:8080/user/admin/notify
```

**Comment by gbaz** [comment]
@peterbecich any thoughts on why the nix flake test is consistently failing?

**Comment by ulysses4ever** [comment]
RE nix failure: it's due to Cabal update which didn't take into account the nix setup (#1186). I tried to fix it (https://github.com/ulysses4ever/hackage-server/commit/aa00808c8cdecf32745031481ec91395e15eeeb4) but bumped into a test suite failure (https://github.com/ulysses4ever/hackage-server/actions/runs/4682410347/jobs/8296230988) and i didn't go as far as figuring what's the best way to handle it (e.g. one could probably disable the tests).

**Comment by peterbecich** [comment]
Will look at the Nix error later today. Have you tried `nix flake update`? Thanks

**Comment by peterbecich** [comment]
Attempting to fix: https://github.com/haskell/hackage-server/pull/1201

**Comment by ysangkok** [comment]
@gbaz I have now double checked everything and I think this is ready for merging!

**Pull request merged**